src/SearchBook.js

```diff
@@ -0,0 +1,90 @@
+import React from "react"
+import {Link} from "react-router-dom"
+import Book from "./Book"
+
+import * as BooksAPI from './BooksAPI'
+
+class SearchBook extends React.Component {
+  state = {
+    enteredText: '',
+    queriedBooks: [],
+    myBooks: []
+  }
+  componentDidMount() {
+    BooksAPI.getAll().then(myBooks => {
+      this.setState({ myBooks })
+    })
+  }
+  handleChange = (enteredText) => {
+    this.setState(() => {return {enteredText: enteredText}})
+    this.search_books(enteredText)
+  }
+  search_books = (serchText) => {
+    if(serchText) {
```

Comment:
It would make it easier to read this if you found a ways to do it without nested `if...else` blocks.

Reply:
+